ios/ImagePickerManager.m

```diff
@@ -89,17 +89,14 @@ - (void)imagePickerController:(UIImagePickerController *)picker didFinishPicking
 
         if ([info[UIImagePickerControllerMediaType] isEqualToString:(NSString *) kUTTypeImage]) {
             UIImage *image = [ImagePickerManager getUIImageFromInfo:info];
-            [assets addObject:[self mapImageToAsset:image data:nil]];
+            [assets addObject:[self mapImageToAsset:image data:[NSData dataWithContentsOfURL:[ImagePickerManager getNSURLFromInfo:info]]]];
         } else {
             NSDictionary *asset = [self mapVideoToAsset:info[UIImagePickerControllerMediaURL]];
-            if (asset == nil) return;
```

Comment:
mapVideoToAsset can fail. in that case it will have called the callback already with the error.

now, it will add a nil video to the assets array and later call the callback again (which will throw because it can only be called once)

Reply:
Ya, just now checked then we must refactor that function to throw error instead of calling callback and catch it where we call.

Reply:
indeed, seems good. didnt want to include that many changes in my original PR.

Reply:
refactored